src/libraries/Common/src/Interop/Browser/Interop.JavaScript.JSObject.cs

```diff
@@ -0,0 +1,166 @@
+                var res = Runtime.InvokeJSWithArgs(JSHandle, method, args, out int exception);
+                if (exception != 0)
+                    throw new JSException((string)res);
+                return res;
+            }
+
+            public object GetObjectProperty(string name)
+            {
+
+                var propertyValue = Runtime.GetObjectProperty(JSHandle, name, out int exception);
+
+                if (exception != 0)
+                    throw new JSException((string)propertyValue);
+
+                return propertyValue;
+
+            }
+
+            public void SetObjectProperty(string name, object value, bool createIfNotExists = true, bool hasOwnProperty = false)
+            {
+
+                var setPropResult = Runtime.SetObjectProperty(JSHandle, name, value, createIfNotExists, hasOwnProperty, out int exception);
+                if (exception != 0)
+                    throw new JSException($"Error setting {name} on (js-obj js '{JSHandle}' mono '{(IntPtr)Handle} raw '{RawObject != null})");
+
+            }
+
+            /// <summary>
+            /// Gets or sets the length.
+            /// </summary>
+            /// <value>The length.</value>
+            public int Length
+            {
+                get => Convert.ToInt32(GetObjectProperty("length"));
+                set => SetObjectProperty("length", value, false);
+            }
+
+            /// <summary>
+            /// Returns a boolean indicating whether the object has the specified property as its own property (as opposed to inheriting it).
+            /// </summary>
+            /// <returns><c>true</c>, if the object has the specified property as own property, <c>false</c> otherwise.</returns>
+            /// <param name="prop">The String name or Symbol of the property to test.</param>
+            public bool HasOwnProperty(string prop) => (bool)Invoke("hasOwnProperty", prop);
+
+            /// <summary>
+            /// Returns a boolean indicating whether the specified property is enumerable.
+            /// </summary>
+            /// <returns><c>true</c>, if the specified property is enumerable, <c>false</c> otherwise.</returns>
+            /// <param name="prop">The String name or Symbol of the property to test.</param>
+            public bool PropertyIsEnumerable(string prop) => (bool)Invoke("propertyIsEnumerable", prop);
+
+            protected void FreeHandle()
+            {
+                Runtime.ReleaseHandle(JSHandle, out int exception);
+                if (exception != 0)
+                    throw new JSException($"Error releasing handle on (js-obj js '{JSHandle}' mono '{(IntPtr)Handle} raw '{RawObject != null})");
+            }
+
+            public override bool Equals(object? obj)
+            {
+                if (obj == null || GetType() != obj.GetType())
+                {
+                    return false;
+                }
+                return JSHandle == (obj as JSObject)?.JSHandle;
```

Comment:
Is the GetType check necessary?  Just wondering if this could be:
```C#
public override bool Equals(object? obj) => obj is JSObject other && JSHandle == other.JSHandle;
```

Reply:
Addressed in recent commit